protocols/luci-proto-yggdrasil/htdocs/luci-static/resources/protocol/yggdrasil.js

```diff
@@ -0,0 +1,272 @@
+'use strict';
+'require form';
+'require network';
+'require rpc';
+'require tools.widgets as widgets';
+'require uci';
+'require ui';
+network.registerPatternVirtual(/^yggdrasil-.+$/);
+
+function validatePrivateKey(section_id,value) {
+	if (value.length == 0) {
+		return true;
+	};
+	if (!value.match(/^([0-9a-fA-F]){128}$/)) {
+		if (value != "auto") {
+			return _('Invalid private key string');
+		}
+		return true;
+	}
+	return true;
+};
+
+function validatePublicKey(section_id,value) {
+	if (value.length == 0) {
+		return true;
+	};
+	if (!value.match(/^([0-9a-fA-F]){64}$/))
+		return _('Invalid public key string');
+	return true;
+};
+
+function validateYggdrasilListenUri(section_id,value) {
+	if (value.length == 0) {
+		return true;
+	};
+	if (!value.match(/^(tls|tcp|unix|quic):\/\//))
+		return _('URI scheme not supported');
```

Comment:
Quote `value` in the return string so we know which URI was problematic: multiple URIs are possible.

Reply:
e.g. 
```js
 return _('unsupported URI scheme in ' + value);
```

Reply:
or...
e.g. 
```js
 return _('Unsupported URI scheme in %s').format(value);
```

Reply:
ok i will have to test with new examples, but prior to this i swear i saw something like:  

:small_red_triangle: URI scheme not supported  

it was immediatly under any typo'd url field... i'll doublecheck

Reply:
- [ ] add format

Reply:
took the suggested format method -- much appreciated.